api/azure/tfvars.go

```diff
@@ -40,8 +43,24 @@ func (e *Cluster) AddMachine(
 	name string,
 	machine *api.Machine,
 ) (string, error) {
+	maxNameLen := nameSettings().maxNameLen
+	minAutoNameLen := nameSettings().minAutoNameLen
+
 	if name == "" {
-		name = uuid.New().String()
+		name = strings.Replace(uuid.New().String(), "-", "", -1)
+
+		if len(api.NameHelper(&e.config.BaseConfig)+"-"+name) > maxNameLen {
```

Comment:
It would be nice if we could generalize this instead of having to implement the name validation for each provider. Perhaps it's possible rewrite the `NameHelper` so that you input a name and it validates and returns the full name?